org.eclipse.lsp4e.tests.mock/src/org/eclipse/lsp4e/tests/mock/MockLanguageServerMultiRootFolders.java

```diff
@@ -64,7 +64,7 @@ public final class MockLanguageServerMultiRootFolders implements LanguageServer
 	private MockWorkspaceService workspaceService = new MockWorkspaceService(this::buildMaybeDelayedFuture);
 	private InitializeResult initializeResult = new InitializeResult();
 	private long delay = 0;
-	private boolean started;
+	private volatile boolean started;
```

Comment:
why this change?

Reply:
because we have code that runs in other threads which queries the isStarted() method. without the volatile it is not guaranteed that these threads see the current value.